trunk/NDHMS/Routing/module_HYDRO_io.F

```diff
@@ -9317,19 +9317,21 @@ end function get_netcdf_dim
 subroutine readBucket_nhd(infile, numbasns, gw_buck_coeff, gw_buck_exp, &
                 gw_buck_loss, z_max, z_init, LINKID, nhdBuckMask)
     implicit none
-    integer :: numbasns
+    integer, intent(in) :: numbasns
     integer(kind=int64), dimension(numbasns) :: LINKID
     real, dimension(numbasns) :: gw_buck_coeff, gw_buck_exp, gw_buck_loss
     real, dimension(numbasns) :: z_max, z_init
     integer, dimension(numbasns) :: nhdBuckMask
-    character(len=*) :: infile
+    character(len=*), intent(in) :: infile
 !   define temp array
-    integer :: i,j,k, gnid, ncid, varid, ierr, dimid, iret
+    integer, volatile :: i,k
```

Comment:
I just realized we can change these integers to volatile for only PGI compilers by sneaking in a simple macro. Helps with portability. For instance
```
integer &
#ifdef __PGI
   , volatile &
#endif
      :: i,k, gnid, ncid, varid, dimid, ierr, iret
```

Reply:
I've found that Cray Fortran often needs the `volatile` attribute as well -- is there any harm to having these here for all compilers?

> I just realized we can change these integers to volatile for only PGI compilers by sneaking in a simple macro. Helps with portability. For instance
> 
> ```
> integer &
> #ifdef __PGI
>    , volatile &
> #endif
>       :: i,k, gnid, ncid, varid, dimid, ierr, iret
> ```